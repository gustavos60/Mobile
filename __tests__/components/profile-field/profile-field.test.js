import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ProfileField from '../../../src/components/profile-field';

const mockedProps = {
  label: 'Nome',
  value: 'Gustavo',
  placeholder: 'Nome',
};

describe('ProfileField tests', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<ProfileField {...mockedProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should hide text input when not in edit mode', () => {
    const wrapper = shallow(<ProfileField {...mockedProps} canEdit={false} />);
    const TextInput = wrapper.find('TextInput').first();
    expect(TextInput.exists()).toBeFalsy();
  });

  it('should show text input when in edit mode', () => {
    const wrapper = shallow(<ProfileField {...mockedProps} canEdit />);
    const TextInput = wrapper.find('TextInput').first();
    expect(TextInput.exists()).toBeTruthy();
  });
});
