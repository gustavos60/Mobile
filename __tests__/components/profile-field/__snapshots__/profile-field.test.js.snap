// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileField tests renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Nome
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    Gustavo
  </Text>
</View>
`;
